src/colvar_rotation_derivative.h

```diff
@@ -327,12 +337,13 @@ struct rotation_derivative {
     *  @param[out] dq0_out The output of derivative of Q
     *  @param[out] ds_out  The output of derivative of overlap matrix S
     */
+  template <bool use_dl, bool use_dq, bool use_ds>
```

Comment:
In practice, the only instances are <false, true, false> and <true, true, true>, so in the current state of the code, this could work with a single boolean template parameter (that could be named `use_dl_ds`). Is this setup for generality, for clarity, or for future-proofing the code?

Reply:
Indeed, after revising the rotation derivative code I find that there are only two combinations used. `use_ds` is mainly used for individually debugging the derivative of overlapping matrix $S$ with respect to atom positions ($\mathrm{d}S/\mathrm{d}\mathbf{x}$), or more specifically, it is used for copying the derivative of overlapping matrix $S$ to a buffer provided by the caller.

`use_dl` means using the derivative of eigenvalues with respect to atom positions, which is actually $\mathrm{d}l/\mathrm{d}S\cdot\mathrm{d}S/\mathrm{d}\mathbf{x}$, so it also uses the $\mathrm{d}S/\mathrm{d}\mathbf{x}$ internally, but I cannot rule out the possibility of using the eigenvalues directly in a CV, and in such case, we may need to have `<true, false, false>`. As a result, I incline to keep the template parameters in the current form.

Reply:
That works for me!